### DISCUSSION TITLE
Maximum number of reminders in an Orleans cluster

### DISCUSSION THREAD

**Discussion by jurby** [dotnet/orleans#8495] [Q&A] [answered]
Hello,

I'm working on a project that utilizes the Orleans framework and I have a question about the use of Orleans reminders. Could you please inform me about the maximum number of reminders that can be held in a single Orleans cluster? Are there any limitations or recommendations regarding the number of reminders that I should keep in mind when designing and implementing our system?

Discord: https://discord.com/channels/333727978460676096/333727978460676096/1119580844714106880

Thank you for your help.

Best Regards,

Jurby

**Comment by ReubenBond** [comment, marked as answer]
There is no fixed maximum, but currently all reminders are loaded and kept in memory, spread around the cluster. So your cluster needs enough memory to hold them all. It is something which we are intending to address in future. I am hoping we can address that for 8.0 so that reminders scale further. Relevant issues: https://github.com/dotnet/orleans/issues/947 & https://github.com/dotnet/orleans/issues/7477
